tests/test_labelarray.py

```diff
@@ -337,3 +339,68 @@ def test_setitem_array(self):
         # Write the whole array.
         arr[:] = orig_arr
         check_arrays(arr, orig_arr)
+
+    def test_narrow_code_storage(self):
+        def create_categories(width, plus_one):
+            length = int(width / 8) + plus_one
+            return [
+                ''.join(cs)
+                for cs in take(
+                    2 ** width + plus_one,
+                    product([chr(c) for c in range(256)], repeat=length),
+                )
+            ]
+
+        # uint8
+        categories = create_categories(8, plus_one=False)
+        arr = LabelArray(
+            [],
+            missing_value=categories[0],
+            categories=categories,
+        )
+        self.assertEqual(arr.itemsize, 1)
+
+        # uint8 inference
```

Comment:
I'm saying we might be incorrectly calculating the cutoffs in this test and in our code, which would mean we're asserting that the size should be 2, and it is 2, but it should have been 4.